lib/matplotlib/style/core.py

```diff
@@ -84,6 +91,101 @@ def use(style):
                 raise IOError(msg % style)
 
 
+def get_parents_styles(styles):
+    """ Returns a list of all styles with parent styles included
+    Parameters
+    ----------
+    styles = list of str and/or dict
+    Example
+    -------
+    Let us consider multiple stylesheet calling each other
+    main
+    |
+    |- a
+    |  |- a1
+    |  |- a2
+    |
+    |- b
+    |  |- b1
+    |  |- b2
+    |
+    |- c
+       |- c1
+       |- c2
+    This means that `main` calls `a`, `b`, and `c`
+    Similar pattern for a, b and c.
+    When calling `matplotlib.style.use(['main']), we will call all the
+    stylesheets recursively add apply all styles. In this example, it would
+    return the following list
+    ['c2', 'c1', 'c', 'b2', 'b1', 'b', 'a2', 'a1', 'a', 'main']
+    """
+    styles = [styles]
+    has_parent = True
+    full_styles = []
```

Comment:
It looks like you're using this to detect cycles. I would call this `visited_styles`.